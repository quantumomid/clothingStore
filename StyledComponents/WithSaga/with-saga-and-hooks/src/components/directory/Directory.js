import React from 'react'
import MenuItem from '../menu-item/MenuItem'
import { useSelector } from 'react-redux'
import { selectDirectorySections } from '../../redux/directory/directorySelector'
import { DirectoryMenuContainer } from './directoryStyles'

const Directory = () => {   
    const sections = useSelector(selectDirectorySections);
    const menuItems = sections.map(({ id, ...otherSectionProps }) => <MenuItem key={id} {...otherSectionProps} />)

    return(
        <DirectoryMenuContainer>
            {menuItems}
        </DirectoryMenuContainer>
    )
}

export default Directory;