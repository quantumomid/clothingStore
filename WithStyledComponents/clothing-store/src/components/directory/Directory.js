import React from 'react'
import MenuItem from '../menu-item/MenuItem'
// import './Directory.scss'
import { connect } from 'react-redux'
import { selectDirectorySections } from '../../redux/directory/directorySelector'
import { createStructuredSelector } from 'reselect'
import { DirectoryMenuContainer } from './directoryMenuContainer'

const Directory = ({ sections }) => {   

    const menuItems = sections.map(({ id, ...otherSectionProps }) => <MenuItem key={id} {...otherSectionProps} />)

    return(
        <DirectoryMenuContainer>
            {menuItems}
        </DirectoryMenuContainer>
    )
}

//OLD version - essentially returns an object that becomes part of the props
// const mapStateToProps = (state) => ({
//     sections: selectDirectorySections(state)
// });

const mapStateToProps = createStructuredSelector({
    sections: selectDirectorySections
});

export default connect(mapStateToProps)(Directory);