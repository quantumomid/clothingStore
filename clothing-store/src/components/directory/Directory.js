import React from 'react'
import MenuItem from '../menu-item/MenuItem'
import './Directory.scss'
import sections from '../../data/directoryData.js'

export default class Directory extends React.Component{
    constructor(){
        super()
        this.state={
            sections: sections
        }
    }

    render(){

        const menuItems = this.state.sections.map(({title}) => {
            return (
                <MenuItem title={title}/>
            )
        })

        return(
            <div className="directory-menu">
                {menuItems}
            </div>
        )
    }
}