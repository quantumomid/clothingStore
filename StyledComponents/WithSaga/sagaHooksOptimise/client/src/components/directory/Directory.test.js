import React from "react";
import { shallow } from "enzyme";
import * as redux from 'react-redux';
import Directory from "./Directory";
import sections from "../../data/directoryData";


describe("Testing the Directory Component", () => {
    let wrapper;
    let spyOnUseSelector;

    beforeEach(() => {
        // Mock useSelector hook
        spyOnUseSelector = jest.spyOn(redux, 'useSelector');
        spyOnUseSelector.mockReturnValue(sections);

        wrapper = shallow(<Directory />)
    })

    afterEach(() => {
        jest.restoreAllMocks();
    });


    it("Directory Component should render", () => {
        expect(wrapper).toMatchSnapshot();
    })
})