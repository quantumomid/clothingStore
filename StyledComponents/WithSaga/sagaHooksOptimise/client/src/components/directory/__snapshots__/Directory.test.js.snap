// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing the Directory Component Directory Component should render 1`] = `ShallowWrapper {}`;
